### PR TITLE
docs(network): redesign transactions to include inputs

### PR DESCRIPTION
**Pull request by mariocao**
This PR includes:

- add signatures to transaction claims, as it is necessary to avoid that transactions could be malleable
- redesign transaction data structure to include additional input fields and removes some unnecessary claim fields

### CODE REVIEW

**Comment by aesedepece** [comment]
General comments:
- The name _claims_ no longer make sense as this field no longer accommodates any kind of response to challenges but only signatures. I can't see why wouldn't this be called _signatures_ now*. 
- Inputs are no longer outpoints. They now couple (1) an outpoint to (2) the _response to the spending conditions sported by the output referred by the outpoint_. 
- Be careful when saying `spending conditions`. That's the term for the explicit or implicit script found in the outputs, not the elements included in inputs for fulfilling the unlock of the funds in those inputs. Therefore _spending conditions_ is synonym to _spending requirements_ or _spending challenge_. This is Bitcoin's _scriptPubKey_.
- The second element in each input, which tries to satisfy the spending conditions, doesn't have a name now, but I think we can reuse _claims_ or come up with a new name. This is equivalent to Bitcoin's _scriptSig_.
- I don't think we should speak about different input types. They are all the same type, but they just try to satisfy the requirements of different output types. Therefore, just like before, it's actually the _claims_ (the second element inside each input) which have different types, each matching an output type.  

<sub>* I know, I know, because of P2SH. But that's simply not a concern for us right now.</sub>

File: docs/protocol/network/data-structures/transaction.md [hunk 1]
```diff
@@ -2,24 +2,58 @@
...
-| `version` |   `u32`    | The transaction data format version number  |
-| `inputs`  | `[input]`  | A list of transaction inputs                |
-| `outputs` | `[output]` | A list of 1 or more transaction outputs     |
-| `claims`  | `[claim]`  | A list of claims (i.e. spending conditions) |
+| Field     |    Type    | Description                                |
+| --------- | :--------: | ------------------------------------------ |
+| `version` |   `u32`    | The transaction data format version number |
+| `inputs`  | `[input]`  | A list of transaction inputs               |
+| `outputs` | `[output]` | A list of 1 or more transaction outputs    |
+| `claims`  | `[claim]`  | A list of claims                           |
```
Comment (girazoki) [inline docs/protocol/network/data-structures/transaction.md]: Might we want to change claims for signatures here? I find the word ambiguous

File: docs/protocol/network/data-structures/transaction.md [hunk 2]
```diff
@@ -85,59 +120,45 @@ commit_value = (data_request_value / witnesses) - commit_fee
...
-| `poe`        | `[u8]` | Proof of eligibility                  |
-| `public_key` | `[u8]` | Public Key used for computing the PoE |
-
-### Commit claim (**reveal stage**)
+```math
+reveal_value = data_request_value - committers * commit_fee - revealers * (reveal_fee + tally_fee + reward)
+```
 
-A commit claim is used to prove that the witness has a valid `reveal`, i.e. the signed data request result and nonce match the previous commitment.
+## Claims
```
Comment (girazoki) [inline docs/protocol/network/data-structures/transaction.md]: Same as before, I would just call them signatures. These they can serve as claims (in the cases of Value Transfer Outputs). I feel the name is confusing

**Comment by aesedepece** [comment]
Merged as 5139a0cf86e41f5b79250a4a08e31f84684ea5d8 and 3313136c458b4e01c3c1e46e567b037ee8157fe9.

**Pull request closed without merging**